orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/SqlExecutionRepository.kt

```diff
@@ -427,6 +451,130 @@ class SqlExecutionRepository(
       )
     }
 
+  override fun retrievePipelineConfigIdsForApplication(application: String): List<String> =
+    withPool(poolName) {
+      return jooq.selectDistinct(field("config_id"))
+        .from(PIPELINE.tableName)
+        .where(field("application").eq(application))
+        .fetch(0, String::class.java)
+    }
+
+  /**
+   * this function supports the following ExecutionCriteria currently:
+   * 'limit', a.k.a page size and
+   * 'statuses'.
+   *
+   * It executes the following query to determine how many pipeline executions exist that satisfy the above
+   * ExecutionCriteria. It then returns a list of all these execution ids.
+   *
+   * It does this by executing the following query:
+   * - If the execution criteria does not contain any statuses:
+   *    SELECT config_id, id
+        FROM pipelines force index (`pipeline_application_idx`)
+        WHERE application = "myapp"
+        ORDER BY
+        config_id;
+   * - If the execution criteria contains statuses:
+   *    SELECT config_id, id
+        FROM pipelines force index (`pipeline_application_status_starttime_idx`)
+        WHERE (
+          application = "myapp" and
+          status in ("status1", "status2)
+        )
+        ORDER BY
+        config_id;
+
+   * It then applies the limit execution criteria on the result set obtained above. We observed load issues in the db
+   * when running a query where the limit was calculated in the query itself. Therefore, we are moving that logic to
+   * the code below to ease the burden on the db in such circumstances.
+   */
+  override fun retrieveAndFilterPipelineExecutionIdsForApplication(
+    application: String,
+    pipelineConfigIds: List<String>,
+    criteria: ExecutionCriteria
+  ): List<String> {
+
+    // baseQueryPredicate for the flow where there are no statuses in the execution criteria
+    var baseQueryPredicate = field("application").eq(application)
+      .and(field("config_id").`in`(*pipelineConfigIds.toTypedArray()))
+
+    var table = if (jooq.dialect() == SQLDialect.MYSQL) PIPELINE.tableName.forceIndex("pipeline_application_idx")
+    else PIPELINE.tableName
+    // baseQueryPredicate for the flow with statuses
+    if (criteria.statuses.isNotEmpty() && criteria.statuses.size != ExecutionStatus.values().size) {
+      val statusStrings = criteria.statuses.map { it.toString() }
+      baseQueryPredicate = baseQueryPredicate
+        .and(field("status").`in`(*statusStrings.toTypedArray()))
+
+      table = if (jooq.dialect() == SQLDialect.MYSQL) PIPELINE.tableName.forceIndex("pipeline_application_status_starttime_idx")
+      else PIPELINE.tableName
+    }
+
+    val finalResult: MutableList<String> = mutableListOf()
+
+    log.info("getting execution ids")
```

Comment:
NIT:  Log.info which isn't terribly useful here.  If anything maybe a log.debug level possibly with something more useful?  OR remove entirely as this I'd think would get called a LOT